infra/recipes/terraform/azure/environments/sandbox/main.tf

```diff
@@ -1,35 +1,82 @@
-resource "azurerm_resource_group" "openhouse_sandbox" {
+data "azurerm_resource_group" "openhouse_sandbox" {
     name = var.resource_group_name
-    location = var.resource_group_location
+}
+
+resource "random_string" "storage_name" {
+    length = 5
+    special = false
+    lower = true
+    upper = false
+}
+
+locals {
+    storage_account_name = "openhousestorage${random_string.storage_name.result}" // added random string of numbers to make it unique
+    container_name = "blobcontainer"
+    db_username = "azureadmin"
+    db_password = "Pa33word"
+    db_name = "openhouse-sandbox-db"
+    db_server_name = "openhouse-sandbox-mysql-server"
 }
 
 module "vm" {
     source = "../../modules/vm"
+    depends_on = [ data.azurerm_resource_group.openhouse_sandbox ]
     virtual_network_name = "openhouse-sandbox-network"
     resource_group_name = var.resource_group_name
     subnet_name = "openhouse-sandbox-subnet"
 }
 
 module "mysql" {
     source = "../../modules/mysql"
+    depends_on = [ module.vm ]
     subnet_id = module.vm.subnet_id
-    resource_group_name = azurerm_resource_group.openhouse_sandbox.name
-    server_name = "openhouse-sandbox-mysql-server"
-    db_admin_login = "azureadmin"
-    db_admin_password = "Pa33word"
-    db_name = "openhouse-sandbox-db"
+    resource_group_name = data.azurerm_resource_group.openhouse_sandbox.name
+    server_name = local.db_server_name
+    db_admin_login = local.db_username
+    db_admin_password = local.db_password
+    db_name = local.db_name
+    server_sku = "B_Standard_B2ms"
 }
 
 module "k8s" {
   source = "../../modules/k8s"
-  k8s_cluster_name = "openhouse-sandbox-k8s"
-  resource_group_name = azurerm_resource_group.openhouse_sandbox.name
-  node_count = 1
-  vm_size = "Standard_D2s_v3"
+  depends_on = [ data.azurerm_resource_group.openhouse_sandbox ]
+  k8s_cluster_name = var.k8s_cluster_name
+  resource_group_name = data.azurerm_resource_group.openhouse_sandbox.name
+  min_node_count = 2
+  max_node_count = 5
+  vm_size = "Standard_B2pls_v2"
+  acr_id = data.azurerm_container_registry.default.id
 }
 
 module "storage" {
     source = "../../modules/storage"
-    storage_account_name = "openhousestorage729387" // added random string of numbers to make it unique
-    resource_group_name = azurerm_resource_group.openhouse_sandbox.name
-}
\ No newline at end of file
+    depends_on = [ data.azurerm_resource_group.openhouse_sandbox ]
```

Comment:
can we introduce storage in a future PR when storageType is changed from local -> abs?

Reply:
Storage was already introduced in the previous PR as a module. Should I just remove the module call from sandbox main.tf for now and then add it again when I add the adls storage?